torchrec/distributed/planner/planners.py

```diff
@@ -302,3 +300,267 @@ def plan(
+        shards = _get_parameter_shards(
+            param,
+            ShardingType.TABLE_ROW_WISE,
+            local_size,
+            world_size,
+        )
+
+        assert len(shards) <= local_size
+        for shard, rank in zip(shards, range(local_size)):
+            shard.rank = node_index + rank
+
+        return _get_parameter_sharding(
+            ShardingType.TABLE_ROW_WISE.value, shards, local_size, device_type
+        )
+
+    return _parameter_sharding_generator
+
+
+def construct_parameter_sharding_plans(
+    module: nn.Module,
+    per_param_sharding: Dict[
+        str, Callable[[nn.Parameter, int, int, str], ParameterSharding]
+    ],
+    sharder: ModuleSharder[nn.Module],
+    local_size: Optional[int] = None,
+    world_size: Optional[int] = None,
+    device_type: str = "cuda",
+) -> Dict[str, ParameterSharding]:
+    """
+    Helper function to create sharding plans (Dict[str, ParameterSharding]) for an module
+    Args:
+        module (nn.Module): module to create plan for.
+        per_param_sharding: Dict[str, Callable[[nn.Parameter, int, int, str], ParameterSharding]]: A mapping of parameter names to a generator function
+        that takes in [parameter, local_size, world_size, device_type] and returns a ParameterSharding. We recommend using one of the predefined generator functions
+        e.g. table_wise_sharding, row_wise_sharding, etc,
+        sharder: ModuleSharder[nn.Module]: Sharder that we are creating a plan for,
+        local_size: Optional[int] = None: process group local size
+        world_size: Optional[int] = None: process_group world_size
+        device_type: str : Torch device type,
+
+    Example::
+        ebc = EmbeddingBagCollection(...)
+        plan = construct_parameter_sharding_plan(
+            ebc,
+            {
+                "table_0": data_parallel_sharding(),
+                "table_1": row_wise_sharding(),
+                "table_2": column_wise_sharding(),
+                "table_3": column_wise_sharding(ranks=[0,1,2]),
+                "table_4": table_row_wise_sharding(host_index=2),
+            },
+            EmbeddingBagCollectionSharder()
+        )
+    """
+    assert isinstance(
+        module, sharder.module_type
+    ), f"Incorrect sharder for module type {type(module)}"
+    shardable_parameters = sharder.shardable_parameters(module)
+    assert (
+        shardable_parameters.keys() == per_param_sharding.keys()
+    ), "per_param_sharding_config doesn't match the shardable parameters of the module"
+
+    local_size = local_size or get_local_size()
+    world_size = world_size or dist.get_world_size()
+
+    per_parameter_sharding: Dict[str, ParameterSharding] = {}
+
+    for param_name, sharding_plan_generator in per_param_sharding.items():
+        param = None
+        for name, _param in module.named_parameters():
+            if param_name in name:
+                param = _param
+                break
+        assert param is not None
```

Comment:
better error message

Reply:
the param_name in name is also a bit annoying because we strip off ".weight", I guess I can get substring bugs